GVRf/Framework/framework/src/main/java/org/gearvrf/GVRLightBase.java

```diff
@@ -116,29 +116,10 @@ static public long getComponentType() {
      */
     public void setCastShadow(boolean enableFlag)
     {
-        GVRSceneObject owner = getOwnerObject();
-
-        if (owner != null)
+        if (enableFlag)
```

Comment:
It looks like GVRLightBase could be an abstract class with no knowledge of shadows. Would make GVRLightBase and GVRPointLight slightly simpler and the hierarchy clearer.